backend/app/routers/garmin_courses.py

```
@@ -138,6 +138,8 @@ async def get_city_coordinates(city: str = Query(...), state: str = None, countr
             return {"latitude": location.latitude, "longitude": location.longitude}
         else:
             raise HTTPException(status_code=404, detail="Location not found")
+    except HTTPException:
+        raise
     except Exception as e:
         raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.15.11)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)